x-pack/packages/security-solution/side_nav/src/solution_side_nav.tsx

```diff
@@ -168,102 +127,251 @@ export const SolutionSideNav: React.FC<SolutionSideNavProps> = React.memo(functi
-              extraAction: {
-                className: buttonClassNames,
-                color: isActive ? 'primary' : 'text',
-                onClick: onButtonClick,
-                iconType: 'spaces',
-                iconSize: 'm',
-                'aria-label': 'Toggle panel nav',
-                'data-test-subj': `solutionSideNavItemButton-${id}`,
-                alwaysShow: true,
-              },
-            })}
-          />
-        </EuiLink>
-        {appendSeparator && <EuiHorizontalRule margin="xs" />}
+        <EuiFlexGroup alignItems="center" gutterSize="xs">
+          <EuiFlexItem>
+            <EuiListGroupItem
+              label={itemLabel}
+              href={href}
+              wrapText
+              onClick={onLinkClicked}
+              className={itemClassNames}
+              color="text"
+              size="s"
+              data-test-subj={`solutionSideNavItemLink-${id}`}
+            />
+          </EuiFlexItem>
+          {hasPanelNav && (
+            <EuiFlexItem grow={0}>
+              <EuiButtonIcon
+                className={buttonClassNames}
+                display={isActive ? 'base' : 'empty'}
+                size="s"
+                color="text"
+                onClick={onButtonClick}
+                iconType="spaces"
+                iconSize="m"
+                aria-label="Toggle panel nav"
+                data-test-subj={`solutionSideNavItemButton-${id}`}
+              />
+            </EuiFlexItem>
+          )}
+        </EuiFlexGroup>
+        {appendSeparator ? <EuiHorizontalRule margin="xs" /> : <EuiSpacer size="xs" />}
       </>
     );
   }
 );
 
+interface SolutionSideNavPanelsProps {
+  items: SolutionSideNavItem[];
+  activePanelNavId: ActivePanelNav;
+  onClose: () => void;
+  onOutsideClick: () => void;
+  bottomOffset?: string;
+  topOffset?: string;
+}
+type NavItemsById = Record<
+  string,
+  {
+    title: string;
+    panelItems: SolutionSideNavItem[];
+    categories?: LinkCategories;
+  }
+>;
+/**
+ * The Solution side navigation panels component.
+ * Renders the proper the secondary panel according to the `activePanelNavId` received.
+ */
+const SolutionSideNavPanels: React.FC<SolutionSideNavPanelsProps> = React.memo(
+  function SolutionSideNavPanels({
+    items,
+    activePanelNavId,
+    onClose,
+    onOutsideClick,
+    bottomOffset,
+    topOffset,
+  }) {
+    const navItemsById = useMemo<NavItemsById>(
+      () =>
+        items.reduce<NavItemsById>((acc, navItem) => {
+          if (navItem.items?.length) {
+            acc[navItem.id] = {
+              title: navItem.label,
+              panelItems: navItem.items,
+              categories: navItem.categories,
+            };
+          }
+          return acc;
+        }, {}),
+      [items]
+    );
+
+    if (activePanelNavId == null || !navItemsById[activePanelNavId]) {
+      return null;
+    }
+    const { panelItems, title, categories } = navItemsById[activePanelNavId];
```

Comment:
Since you are only using one property of `navItemsById`, does it make more sense to find the `navItem` by id instead of doing a `reduce` and a lookup?

Something like:
```suggestion
    const item = items.find((item) => item.id === activePanelNavId);
     if (activePanelNavId == null || !item) {
      return null;
    }
    const { panelItems, title, categories } = item;
```

Reply:
The `navItemsById` was passed to the children in the previous implementation, I kept it because getting the panel by key is more performant than iterating the `items` array every time the `activePanelNavId` changes. But we won't have many items here, so I guess it's just fine to use `find`.